cdap-api/src/main/java/io/cdap/cdap/api/metrics/Metrics.java

```diff
@@ -34,5 +36,17 @@
    * @param value The value to be set.
    */
   void gauge(String metricName, long value);
+
+  /**
+   * Get the child {@link Metrics} by adding specified tags to the current metrics context.
```

Comment:
what is the behavior if the tags here overlap with the tags for the current Metrics instance?

Reply:
Implementation of this method delegates to the MetricsContext implementation. Current `MetricsContext` implementation has odd behavior though - 
1. If tag is added using this method - https://github.com/cdapio/cdap/blob/develop/cdap-watchdog/src/main/java/io/cdap/cdap/metrics/collect/AggregatedMetricsCollectionService.java#L241 IllegalArgumentException would be thrown for duplicate keys.
2. If tag is added using this method - https://github.com/cdapio/cdap/blob/develop/cdap-watchdog/src/main/java/io/cdap/cdap/metrics/collect/AggregatedMetricsCollectionService.java#L256 duplicate keys are allowed.

Reply:
I think its better to throw exception when overlap happens so that user dont accidentally updates values for tags such as runid, namespace etc. Let me know your thoughts.

Reply:
> I think its better to throw exception when overlap happens so that user dont accidentally updates values for tags such as runid, namespace etc. Let me know your thoughts.

+1, we should make them consistent and to avoid accidentally override the predefined tags, it would better to throw exceptions.

Reply:
Throwing exception in presence of duplicate tags from Metrics API now. Please check.